Compute/tests/System/V1/SmokeTest.php

```diff
@@ -87,30 +104,63 @@ public function testInsertInstance()
             self::$projectId,
             self::ZONE
         );
-        try{
-            $operationClient = new ZoneOperationsClient();
-            while (true) {
-                $op = $operationClient->get(
-                    $operation->getName(),
-                    self::$projectId, self::ZONE
-                );
-                $status = $op->getStatus();
-                if (in_array($status, [Status::DONE, Status::UNDEFINED_STATUS])) {
-                    break;
-                }
-            }
-            $instance = self::$instancesClient->get(
-                $name,
-                self::$projectId,
-                self::ZONE
-            );
-        } finally {
-            self::$instancesClient->delete($name, self::$projectId, self::ZONE);
-        }
-        self::assertEquals($name, $instance->getName());
+        $this->waitForZonalOp($operation);
+        self::$dirty = true;
+    }
+
+    public function testInsertInstance()
+    {
+        $this->insertInstance();
+        $instance = self::$instancesClient->get(
+            self::$name,
+            self::$projectId,
+            self::ZONE
+        );
+        self::assertEquals(self::$name, $instance->getName());
         self::assertEquals(self::$machineType, $instance->getMachineType());
     }
 
+    public function testPatchInstance(){
+        $shieldedInstanceConfigResource = new ShieldedInstanceConfig();
+        $shieldedInstanceConfigResource->setEnableSecureBoot(true);
+        $this->insertInstance();
+        self::$instancesClient->stop(self::$name, self::$projectId, self::ZONE);
+        while (true){
+            $instance = $this->getInstance();
+            if ($instance->getStatus() == Instance\Status::TERMINATED):
+                break;
+            endif;
+            sleep(10);
+        }
+        try {
+            $op = self::$instancesClient->updateShieldedInstanceConfig(
+                self::$name, self::$projectId, $shieldedInstanceConfigResource, self::ZONE);
+        } catch (ApiException $e) {
+            self::fail("update method failed" . $e->getMessage());
+        }
+        $this->waitForZonalOp($op);
+        $instance = $this->getInstance();
+        self::assertEquals(true, $instance->getShieldedInstanceConfig()->getEnableSecureBoot());
+
+    }
+
+    public function waitForZonalOp($operation): void
+    {
+        try {
+            self::$zoneOperationsClient->wait($operation->getName(), self::$projectId, self::ZONE);
+        } catch (ApiException $e) {
+            self::fail("Wait on zonal operation failed" . $e->getMessage());
```

Comment:
use `$this` instead of `self`
```suggestion
            $this->fail("Wait on zonal operation failed" . $e->getMessage());
```